### PR TITLE
add filesystem tests to try to increase coverage

### PR DESCRIPTION
**Pull request by birm**
When looking around, I couldn't help but notice some functions in base/filesystem.jl didn't seem to have [test coverage](https://coveralls.io/builds/28847289/source?filename=base/filesystem.jl).

I'm almost certain that, as is, these aren't particularly good or effective tests, but I thought it was a reasonable starting place. I'm certainly willing to edit with feedback.

### CODE REVIEW

File: test/filesystem.jl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  # test filesystem truncate (lengthen)
+  file = Base.Filesystem.open(filename, Base.Filesystem.JL_O_RDWR)
+  Base.Filesystem.truncate(file, 20)
+  @test length(read(file)) == 20
+  close(file)
+
+  # test filesystem futime
+  file = Base.Filesystem.open(filename, Base.Filesystem.JL_O_RDWR)
+  Base.Filesystem.futime(file, 1.0, 1.0)
+  close(file)
```
Comment (vtjnash) [inline test/filesystem.jl]: You can probably test calling `stat` here to make sure the change was successful

File: test/filesystem.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  # test filesystem readbytes!
+  file = Base.Filesystem.open(filename, Base.Filesystem.JL_O_RDWR)
+  res = zeros(UInt8,20*4)
```
Comment (vtjnash) [inline test/filesystem.jl]:
```suggestion
  res = ones(UInt8, 80)
```

File: test/filesystem.jl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  Base.Filesystem.readbytes!(file, res)
+  res[1] == 0x31
```
Comment (vtjnash) [inline test/filesystem.jl]:
```suggestion
  @test res == UInt8[text..., (i > 20 for i in (length(text) + 1):length(res))...]
```

**Review by vtjnash — COMMENTED** [review]
Yes, these seem like a good starting point!

File: test/filesystem.jl (hunk 1 above)
Comment (birm) [reply test/filesystem.jl]: Good thought; thanks

**Pull request merged**